cmd/config.go

```diff
@@ -249,21 +250,130 @@ func (d *ConfigDuration) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
-// GRPCClientConfig contains the information needed to talk to the gRPC service
+// GRPCClientConfig contains the information setup a gRPC client. The following
+// combination of fields are allowed:
+//
+// ServerIPAddresses, [Timeout]
+// ServerAddress, [Timeout], [DNSAuthority]
+// SRVLookup, [Timeout], [DNSAuthority]
 type GRPCClientConfig struct {
-	// ServerAddress is a single host:port combination that the gRPC client
-	// will, if necessary, resolve via DNS and then connect to. This field
-	// cannot be used in combination with `ServerIPAddresses` field.
+	// DNSAuthority is a single <hostname|IPv4|[IPv6]>:<port> or `:<port>` of
+	// the DNS server to be used for resolution of gRPC backends. If the address
+	// contains a hostname the gRPC client will resolve it via the system DNS.
+	// If the address contains a port, the client will use it directly,
+	// otherwise port 53 is used.
+	DNSAuthority string
+
+	// SRVLookup contains the service and domain name the gRPC client will use
+	// to construct a SRV DNS query to lookup backends. For example: if the
+	// resource record is 'foo.service.consul', then the 'Service' is 'foo' and
+	// the 'Domain' is 'service.consul'. The expected dNSName to be
+	// authenticated in the client certificate would be 'foo.service.consul'.
+	//
+	// Note: The 'proto' field of the SRV record MUST be 'tcp' and the 'port'
+	// field MUST be contain valid port. In a Consul configuration file you
+	// would specify 'foo.service.consul' as:
+	//
+	// services {
+	//   id      = "some-unique-id-1"
+	//   name    = "foo"
+	//   address = "10.77.77.77"
+	//   port    = 8080
+	//   tags    = ["tcp"]
+	// }
+	// services {
+	//   id      = "some-unique-id-2"
+	//   name    = "foo"
+	//   address = "10.88.88.88"
+	//   port    = 8080
+	//   tags    = ["tcp"]
+	// }
+	//
+	// If you've added the above to your Consul configuration file (and reloaded
+	// Consul) then you should be able to resolve the following dig query:
+	//
+	// $ dig @10.55.55.10 -t SRV _foo._tcp.service.consul +short
+	// 10.77.77.77 8080
+	// 10.88.88.88 8080
+	SRVLookup *struct {
```

Comment:
Wow this is neat! I didn't realize you could do this.

Reply:
And yeah, I actually learned about it from these Consul docs: https://www.consul.io/docs/discovery/dns#rfc-2782-lookup

Reply:
Oh 😵‍💫 you're talking about `*struct{`. Yeah I didn't know I could do it until I tried.